packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/EmailInvoiceDetails.tsx

```diff
@@ -72,17 +76,65 @@ const EmailInvoiceDetailsBase = ({
 }: EmailInvoiceDetailsProps) => {
   const { i18n } = useLingui();
   const { monite, api } = useMoniteContext();
-  const { control, handleSubmit, getValues, setValue, trigger } = useForm({
-    resolver: yupResolver(getEmailInvoiceDetailsSchema(i18n)),
-    defaultValues: {
-      subject: '',
-      body: '',
-      to: '',
-      // TODO: add support for multiple recipients, cc and bcc fields
-    },
-  });
+
+  const { control, handleSubmit, getValues, setValue, trigger, watch } =
+    useForm({
+      resolver: yupResolver(getEmailInvoiceDetailsSchema(i18n)),
+      defaultValues: {
+        subject: '',
+        body: '',
+        to: '',
+        // TODO: add support for multiple recipients, cc and bcc fields
+      },
+    });
+
+  const { data: me } = useMe();
+  const { data: receivable } = useReceivableById(invoiceId);
+  const { data: contacts } = useReceivableContacts(invoiceId);
+  const to = watch('to');
+  const body = watch('body');
```

Comment:
I think you can use oneWatch to listen for multiple values to not create extra varriable

Reply:
This won't be needed in the refactors I am introducing.